src/client/features/interactions/index.js

```diff
@@ -56,66 +56,20 @@ class Interactions extends React.Component {
-      source : sources,
-      pattern : ['controls-phosphorylation-of','in-complex-with','controls-expression-of', 'interacts-with'],
-      kind : sources.length>1 ? 'pathsbetween' : 'neighborhood',
-      format : 'txt',
-      //TODO: consider using direction:'bothstream' for neighborhood queries (or remove 'interacts-with' PPI-only pattern)
-      // direction:'bothstream' //ignored if it's pathsbetween
-    };
-    ServerAPI.pcQuery('graph', params)
-      .then(res=>res.text())
-      .then(res=>{
-      const layoutConfig = getLayoutConfig('interactions');
-      const network= this.parse(res);
-      this.setState({
-        componentConfig: interactionsConfig,
-        layoutConfig: layoutConfig,
-        networkJSON: network,
-        loaded:_.assign(this.state.loaded,{network:true})
-      });
-    });
-    //get ids from uris
-    const geneIds = sources.map(source =>
-      source.includes('pathwaycommons')
-        ? ServerAPI.pcQuery('traverse',
-          {
-            uri:source,
-            path:`${_.last(source.split('/')).split('_')[0]}/displayName`
-          })
-          .then(result=>result.json())
-          .then(id=> _.words(id.traverseEntry[0].value[0]).length===1 ? id.traverseEntry[0].value[0].split('_')[0] : '')
-        : source.replace(/\//g,' ')
-    );
-    Promise.all(geneIds).then(geneIds=>{
-      ServerAPI.geneQuery({genes:geneIds,targetDb: 'NCBIGENE'}).then(result=>{
-        const ncbiIds=result.geneInfo.map(gene=> gene.convertedAlias);
-        ServerAPI.getGeneInformation(ncbiIds).then(result=>{
-          const geneResults=result.result;
-          let hgncIds=[];
-          let comments=[];
-          if(!result.esummaryresult ){
-            comments=_.flatten(geneResults.uids.map(gene=>{
-              hgncIds.push(geneResults[gene].name);
-              return _.compact([
-                'Nomenclature Name: '+geneResults[gene].nomenclaturename,
-                'Other Aliases: '+geneResults[gene].name + (geneResults[gene].otheraliases ? ', '+geneResults[gene].otheraliases:''),
-                geneResults[gene].summary && 'Function: '+geneResults[gene].summary
-              ]);
-            }));
-          }
-          this.setState({
-            networkMetadata: {
-              name: hgncIds.length === sources.length ?(hgncIds+' Interactions'):' Interactions',
-              datasource: 'Pathway Commons',
-              comments: comments
-            },
-            ids:hgncIds,
-            loaded:_.assign(this.state.loaded,{ids:true})
-          });
+
+    ServerAPI.getInteractionGraph({sources:sources})
+      .then(result=>{
+        const layoutConfig = getLayoutConfig('interactions');
+        const network= result.network;
+        const metaData = result.metaData;
+        this.setState({
+          componentConfig : interactionsConfig,
+          layoutConfig : layoutConfig,
+          networkJSON : network,
+          networkMetadata : metaData.networkMetadata,
```

Comment:
Let's drop `networkMetadata` and it's creation (see below `src/server/graph-generation/interaction/index.js`).  All of this was intended to populate the  'Info' panel - however, it doesn't add much value, some of the approaches are fragile (traverse), and adds unnecessary bloat.